src/plugins/ui_core_views/custom_colors.ts

```diff
@@ -220,13 +208,15 @@ export class CustomColorsPlugin extends UIPlugin<CustomColorState> {
     ].filter(isDefined);
   }
 
-  private tryToAddColor(color: Color) {
-    if (!isColorValid(color)) {
-      return;
-    }
-    const formattedColor = toHex(color);
-    if (color && !COLOR_PICKER_DEFAULTS.includes(formattedColor)) {
-      this.history.update("customColors", formattedColor, true);
+  private tryToAddColors(colors: Color[]) {
+    for (const color of colors) {
+      if (!isColorValid(color)) {
+        return;
```

Comment:
otherwise, as soon as one color is invalid, all the following colors are ignored
```suggestion
        continue;
```